compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -678,6 +678,148 @@ def IREELinalgExt_AttentionOp : IREELinalgExt_Op<"attention",
+  let summary = "Im2col operation for convolutions";
+  let description = [{
+    Im2col op for convolutions. The operation performs a transformation on the
+    input to convert it from a convolution input to an equivalent gemm input.
+    The op is defined by its input, output, some conv metadata, and some
+    indexing metadata. The `strides`, `dilations`, and `kernel_size` are taken
+    from the convolution from which this op is generated, and they define how
+    the input operand is indexed when the operation is decomposed. The shape of
+    the output should be `tensor<BxMxK>`, and the `m_pos`, `k_pos`, and
+    `batch_pos` indicate which input dimensions map to which output dimensions.
+
+    The `k_offset` is an offset within the output K dimension from which the
+    iteration space of the operation begins. This is used for tiling, since the
+    tiled implementation must leave the output K dimension untiled. Similarly,
+    `m_offset` is the offset within the output M dimension from which the
+    iteration space of the operation begins.
+    The iteration space is the full output shape of the im2col op, so if the
+    im2col op were tiled to loops with a scalar inner tile, it would look like
+    the following:
+    ```
+      %im2col = iree_linalg_ext.im2col
+          strides = [1, 1] dilations = [1, 1] kernel_size = [3, 3]
+          m_offset = [0] k_offset = [0]
+          batch_pos = [0] m_pos = [1, 2] k_pos = [3]
+          ins(%in : tensor<2x34x34x640xf32>)
+          outs(%out : tensor<2x1024x5760xf32>) -> tensor<2x1024x5760xf32>
+    ```
+    becomes:
+    ```
+      scf.for %arg0 = %c0 to %c2 step %c1
+        scf.for %arg1 = %c0 to %c1024 step %c1
+          scf.for %arg2 = %c0 to %c5760 step %c1
+            %im2col = iree_linalg_ext.im2col
+                strides = [1, 1] dilations = [1, 1] kernel_size = [3, 3]
+                m_offset = [%arg1] k_offset = [%arg2]
+                batch_pos = [0] m_pos = [1, 2] k_pos = [3]
+                ins(%in_tile : tensor<1x34x34x640xf32>)
+                outs(%out_tile : tensor<1x1x1xf32>) -> tensor<1x1x1xf32>
+    ```
+    Then, when the tiled op is decomposed, it becomes a loop over the iteration
+    space of the im2col op, whith an extract_slice from the `%in_tile` followed
+    by an insert_slice to the `%out_tile`. The indices for the extract slice are
+    computed using the `m_offset` and `k_offset` as:
+    (b, m, k) -> (b, M / 32 + K / (640*3), M % 32 + K % (640*3) / 640, K % 640)
+    Where `(b, m, k)` are the indices of the tiled op's iteration space, and
+    `M = m + m_offset` and `K = k + K_offset`.
+  }];
+
+  let arguments = (ins Variadic<AnyShaped>:$inputs,
+                       Variadic<AnyShaped>:$outputs,
+                       Variadic<Index>:$strides,
+                       DenseI64ArrayAttr:$static_strides,
+                       Variadic<Index>:$dilations,
+                       DenseI64ArrayAttr:$static_dilations,
```

Comment:
Why do we have dynamic strides and dilations? I assumes that the op will always come from a linalg conv op. IIRC, all the linalg conv ops have constant strides and dilations. E.g.,

https://github.com/llvm/llvm-project/blob/39048b69b85e530b9b8a4226d9043a0bd340fe8a/mlir/include/mlir/Dialect/Linalg/IR/LinalgNamedStructuredOps.yaml#L2571-L2582

Can it just be two `DenseI64ArrayAttr`?